firstvoices/backend/pagination.py

```diff
@@ -1,16 +1,25 @@
 from collections import OrderedDict
 
 from django.core.paginator import InvalidPage, Page, Paginator
+from django.utils.functional import cached_property
 from rest_framework import pagination
 from rest_framework.exceptions import NotFound
 from rest_framework.response import Response
 
 from backend.search.utils.constants import ES_MAX_RESULTS, ES_PAGE_SIZE
 
 
+class FasterCountPagination(Paginator):
+    @cached_property
+    def count(self):
+        # Override the count property to select only the id field to speed up the count
+        return self.object_list.values("id").count()
```

Comment:
I am super into this optimization, and I have a couple of thoughts.

1. This is worth contributing back to Django core.
2. But, this skips some of the safety checks from the original method and they should be added back in. (I believe they are there to make this work for both querysets and regular lists.) If you don't want to test that right now, feel free to log a separate ticket to flesh it out, since we currently only use the paginator for querysets

Reply:
And a question: is there any possibility that doing this produces a different count than the original query, especially if there are table joins without using distinct()? Have you tested it?

Reply:
Some great feedback, thanks!

1. I believe the Django core is this way because it ensures compatibility with all models, not just applications that use a UUID as a primary key.
2. Great point. I will add that in.

I have tested and the sizes come back as the same. The difference between the queries is that the original one is doing the following: 
```
SELECT COUNT(*)
  FROM (
        SELECT DISTINCT "backend_audio"."id" AS "col1",
               "backend_audio"."created" AS "col2",
               "backend_audio"."site_id" AS "col3",
               ...,
          FROM "backend_audio"...
```
and the updated one is doing the following:
```
SELECT COUNT(*)
  FROM (
        SELECT DISTINCT "backend_audio"."id" AS "col1"
          FROM "backend_audio"...
```
The joins on site and membership are the same in both queries.
Since the ID is always unique the count will always be the same.

Reply:
Good point about id. Would "pk" work here instead?